internal/dataplane/translator/subtranslator/grpcroute.go

```diff
@@ -134,11 +136,64 @@ func GenerateKongRoutesFromGRPCRouteRule(
 // getGRPCRouteHostnamesAsSliceOfStringPointers translates the hostnames defined
 // in an GRPCRoute specification into a []*string slice, which is the type required
 // by kong.Route{}.
-func getGRPCRouteHostnamesAsSliceOfStringPointers(grpcroute *gatewayapi.GRPCRoute) []*string {
-	if len(grpcroute.Spec.Hostnames) == 0 {
+// The hostname field is optional. If not specified, the configured value will be obtained from parentRefs.
+func getGRPCRouteHostnamesAsSliceOfStringPointers(grpcroute *gatewayapi.GRPCRoute, storer store.Storer) []*string {
+	hostnames := make([]gatewayapi.Hostname, 0)
+
+	if len(grpcroute.Spec.Hostnames) > 0 {
+		hostnames = grpcroute.Spec.Hostnames
+		return lo.Map(hostnames, func(h gatewayapi.Hostname, _ int) *string {
+			return lo.ToPtr(string(h))
+		})
+	}
```

Comment:
No need to allocate the slice here, if `len(grpcroute.Spec.Hostnames) > 0` then the result of this is discarded

```suggestion
	if len(grpcroute.Spec.Hostnames) > 0 {
		return lo.Map(grpcroute.Spec.Hostnames, func(h gatewayapi.Hostname, _ int) *string {
			return lo.ToPtr(string(h))
		})
	}
```